### PR TITLE
Use asset fetcher for KernelBuild download()

### PR DESCRIPTION
**Pull request by apahim**
Now we have asset fetcher so let's use it.

Reference: https://trello.com/c/d6qeRbJD
Signed-off-by: Amador Pahim <apahim@redhat.com>

### CODE REVIEW

File: avocado/utils/kernel.py [hunk 1]
```diff
@@ -64,12 +64,8 @@ def download(self):
...
         self.kernel_file = self.SOURCE.format(version=self.version)
         full_url = self.URL + self.SOURCE.format(version=self.version)
-        path = os.path.join(self.work_dir, self.kernel_file)
-        if os.path.isfile(path):
-            log.info("File '%s' exists, will not download!", path)
-        else:
-            log.info("Downloading '%s'...", full_url)
-            download.url_download(full_url, path)
+        asset.Asset(full_url, asset_hash=None, algorithm=None,
+                    locations=None, cache_dirs=[self.work_dir]).fetch()
```
Comment (ldoktor) [inline avocado/utils/kernel.py]: Well the point of using `asset` over `download` is the caching, so using `work_dir` is a great drawback. How about accepting optional arguments with datadirs. When not specified the `work_dir` would be used.

Comment (apahim) [inline avocado/utils/kernel.py]: Ok, let me improve this a bit.

Comment (ldoktor) [inline avocado/utils/kernel.py]: Btw how about adding `data_dir` entry for assets to avoid custom handling (eg. inside `Test` class)?

Comment (ldoktor) [inline avocado/utils/kernel.py]: I mean for the use inside the `linuxbuild` test, not directly inside the `utils.kernel`...

**Pull request closed without merging**